components/style/dom_apis.rs

```diff
@@ -275,16 +275,16 @@ where
         return Err(());
     }
 
-    if root.is_in_document() {
-        return root.owner_doc().elements_with_id(id);
-    }
+    if root.is_connected() {
```

Comment:
This change looks totally unrelated, why? It also deoptimizes the case of calling `getElementById` in a disconnected `ShadowRoot`.